twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -169,7 +172,7 @@ public Reader getInput() throws IOException {
   }
 
   private int getReservedMemory() {
-    String value = System.getenv(EnvKeys.TWILL_RESERVED_MEMORY_MB);
+    String value = Integer.toString(twillRuntimeSpec.getReservedMemory());
     if (value == null) {
```

Comment:
This logic doesn't make sense. The `twillRuntimeSpec.getReservedMemory` already return an `int`. It can never be `null`. This method is unnecessary. Just call `twillRuntimeSpec.getReservedMemory` whenever needed.